ecosystem/sep-0038.md

```diff
@@ -234,6 +235,16 @@ Name | Type | Description
 `asset` | string | The [Asset Identification Format](#asset-identification-format) value.
 `price` | string | The price offered by the anchor for one unit of `asset` in terms of `sell_asset`. In traditional finance, `asset` would be referred to as the base asset and `sell_asset` as the counter asset.
 `decimals` | integer | The number of decimals needed to represent `asset`.
+`fee_details` | array | (optional) An array of objects describing the fees that were used to calculate the conversion price. This can be used to datail the price components for the end-user.
```

Comment:
This statement implies that fees are included in the conversion price, but we cannot assume that is how the anchor charges fees. Anchors need to be able to charge fees on top of the amount the client wishes to convert, or subtracted from the amount the client wishes to convert.

This means that the amounts sent in the client's request to the server may be different than the amounts included in the server's response.

I think one of the main benefits of adding this `fee_details` attribute is the ability to describe the difference between what the client requested and what the server provided. If there is no difference in the amounts, which will be the case when the server only includes a margin without fees added/inclusive, I think `fee_details` should be empty.

This allows us to have the following rules:
```
if fee asset == sell_asset:
    sell_amount (response) = sell_amount (request) + fees
else:  // fee asset == buy_asset
    buy_asset (response) = buy_amount (request) - fees
```

The downside of this approach is that it doesn't allow servers to detail their fees if they include them in the margin. I think this is acceptable because I've never seen a service itemize their fees using fixed amounts but collect those amounts using a percentage / margin in the quote.

**Examples**

**Margin-based Fees**
`sell_amount` (from request): 100
`sell_amount` (from response): 100
`buy_amount`: 495
`price`: 0.202 (100 / 495 = 0.202)
`fee`: 0

**Additive / Strict Receive Fees**
`sell_amount` (from request): 100
`sell_amount` (from response): 105
`buy_amount`: 500
`price`: 0.20
`fee_details`: 5 (in units of sell_asset)

**Inclusive / Strict Send Fees**
`sell_amount` (from request): 100
`sell_amount` (from response): 100
`buy_amount`: 495
`price`: 0.2
`fee_details`: 5 (in units of buy_asset)

Reply:
Also, can we use an object for the `fee_details` object?

```json
"fee_details": {
    "asset": "iso4217:BRL",
    "total_amount": "5",
    "details": [
        {
            "name": "Service Fee",
            "description": "...",
            "amount": "5"
        }
    ]
}
```